src/r2mm/manager/SettingsDexieStore.ts

```diff
@@ -24,66 +22,70 @@ export default class SettingsDexieStore extends Dexie {
         });
 
         const store = {
-            value: `++id,settings`
+            value: `++id,settings`,
+            games: `identifier,settings`,
         } as any;
 
-        GameManager.gameList
-            .forEach(value => {
-                store[value.settingsIdentifier] = `++id,settings`;
-            });
+        // Setup the v3 (>= 75) dexie store.
+        // This contains the following tables:
+        // - `games`: (settingsIdentifier,settings) <-- Game-specific settings.
+        // - `value`: (++id, settings)              <-- Global settings.
+        this.version(75).stores(store).upgrade(async (trans) => {
+            // Migrate the current game's legacy (v2) table to the v3 schema, if applicable - versions < 75.
+            //
+            // v2 (whose types we still utilize) stored game-specific settings in individual tables.
+            // This worked but required us to bump the Dexie store version every time we added a new game. Yuck!
+            // To get around this we have moved all game settings into the `games` table, where the key
+            // is the settings identifier of the game, and the value is a JSON string.
+            const ignore = ["value", "games"];
+            const legacyTables = trans.storeNames.filter(x => !ignore.includes(x)).map(x => trans.table(x));
+
+            for (const gameTable of legacyTables) {
+                const legacySettings = await gameTable.toCollection().first();
+
+                // No settings field exists on the game table, skip.
+                if (legacySettings === undefined) {
+                    continue;
+                }
 
-        // Add all games to store. Borked v2-3 locally
-        // Increment per game or change to settings.
-        this.version(74).stores(store);
+                // If the legacy game table exists AND it contains a valid settings field, write it to the games table.
+                // Then clear the contents of the legacy game table.
+                await this.games.put({ identifier: this.activeGame.settingsIdentifier, settings: JSON.stringify(legacySettings.settings) });
```

Comment:
I think at this point `this.activeGame` is always RoR2 so the `identifier` should use some other value, either the old table name or the settingsIdentifier from `GameManager`?